democracy_club/apps/hermes/models.py

```diff
@@ -0,0 +1,280 @@
+
+
+class PostQuerySet(models.query.QuerySet):
+    def by(self, author):
+        return self.filter(author__username=author)
+
+    def in_category(self, category_slug):
+        category = Category.objects.get(slug=category_slug)
+        children = Category.objects.children_of(category)
+
+        return self.filter(category__in=[category] + children)
+
+    def created_on(self, year=None, month=None, day=None):
+        clauses = []
+
+        if year:
+            clauses.append(models.Q(created_on__year=year))
+
+        if month:
+            clauses.append(models.Q(created_on__month=month))
+
+        if day:
+            clauses.append(models.Q(created_on__day=day))
+
+        return self.filter(reduce(operator.__and__, clauses))
+
+    def recent(self, limit=None):
+        queryset = self.published()
+        if limit:
+            queryset = queryset[:limit]
+
+        return queryset
+
+    def random(self, limit=None):
+        queryset = self.recent().order_by("?")
+        if limit:
+            queryset = queryset[:limit]
+
+        return queryset
+
+    def published(self):
+        return self.filter(is_published=True)
+
+    def for_tag(self, tag):
+        return self.filter(tags__contains=[tag])
+
+
+class PostManager(models.Manager):
+    def get_queryset(self):
+        return PostQuerySet(self.model, using=self._db)
+
+    def in_category(self, category_slug):
+        return self.get_queryset().in_category(category_slug)
+
+    def created_on(self, year=None, month=None, day=None):
+        return self.get_queryset().created_on(year=year, month=month, day=day)
+
+    def recent(self, limit=None):
+        return self.get_queryset().recent(limit=limit)
+
+    def random(self, limit=None):
+        return self.get_queryset().random(limit=limit)
+
+    def published(self):
+        return self.get_queryset().published()
+
+    def by(self, author):
+        return self.get_queryset().by(author)
+
+    def for_tag(self, tag):
+        return self.get_queryset().for_tag(tag)
+
+
+def post_hero_upload_to(instance, filename):
+    filename = os.path.split(filename)[-1]
+    filename, extension = os.path.splitext(filename)
+    extension = extension[1:]
+
+    return "hermes/heroes/{slug}_{filename}_hero.{extension}".format(
+        slug=instance.slug[:40], filename=filename[:30], extension=extension
+    )
+
+
+class Post(TimestampedModel):
+    is_published = models.BooleanField(default=False)
+    hero = models.ImageField(
+        _("hero"), upload_to=post_hero_upload_to, blank=True, null=True
+    )
+    subject = models.CharField(_("subject"), max_length=100)
+    slug = models.SlugField(_("slug"), max_length=100)
+    summary = models.TextField(_("summary"), blank=True, null=True)
+    body = models.TextField(_("body"))
+
+    category = models.ForeignKey(Category, on_delete=models.CASCADE)
+    author = models.ManyToManyField(django_settings.AUTH_USER_MODEL)
```

Comment:
`verbose_name="Authors"` here makes the labels in the admin interface read a little nicer